File: 7-kyu/see-you-happy-next-year.js
// Happy Year is the year with only distinct digits , (e.g) 2018

// Given a year, find the next happy year

function nextHappyYear(year){
  year += 1;
  while (!isHappyYear(year)) year += 1;
  return year;
}

function isHappyYear(year) {
  const chars = String(year).split('').sort();

  for (let i = 0; i < chars.length; i += 1) {
    if ((i + 1) === chars.length) {
      return true;
    }

    if (chars[i] === chars[i + 1]) {
      return false;
    }
  }
}

File: 7-kyu/see-you-happy-next-year.clj
;; Happy Year is the year with only distinct digits , (e.g) 2018

;; Given a year, find the next happy year

(ns kata
  (:require [clojure.string :as str]))

(defn is-happy-year? [year]
  (let [str-year (str year)
        length (count str-year)
        uniques (set (str/split str-year #""))]
    (= length (count uniques))))

(defn next-happy-year [year]
  (if (is-happy-year? year) year
      (next-happy-year (inc year))))